corehq/apps/custom_data_fields/tests/test_edit_model.py

```diff
@@ -256,6 +256,36 @@ def _create_form(self, fields, profiles):
         })
 
 
+class TestProfileRequiredForUserType(FieldsViewMixin, TestCase):
+    domain = 'test-domain'
+    BOTH = [UserFieldsView.WEB_USER, UserFieldsView.COMMCARE_USER]
+
+    def test_req_for_defaults_to_none(self):
+        definition = CustomDataFieldsDefinition.objects.create(field_type='UserFields')
+        profiles = [CustomDataFieldsProfile(name='profile', fields={'one': 'one'}, definition=definition)]
+        profiles[0].save()
+        req_for = CustomDataFieldsProfile.objects.get(name='profile').definition.profile_required_for_user_type
+
+        self.assertIsNone(req_for)
+
+    def test_update_profile_required(self):
```

Comment:
Would be good also testing that it can update to BOTH. I remember that was a thing i had issues with in my other PR.

Reply:
https://github.com/dimagi/commcare-hq/pull/35266/commits/7462ab0508400e6139332678cdc36d6f1fa2eae0